scripts/pod_lib_lint.rb

```diff
@@ -42,13 +48,28 @@ def main(args)
 
   command = %w(bundle exec pod lib lint --sources=https://cdn.cocoapods.org/)
 
+  # Split arguments that need to be processed by the script itself and passed
+  # to the pod command.
+  script_args = {}
+  pod_args = []
+  local_specs_to_ignore_arg = ""
+  args.each do |arg|
```

Comment:
Please don't roll your own option parsing. Use `OptionParser` from the standard library. See an example in [sync_project.rb](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/sync_project.rb#L39)

Reply:
@wilhuff Sorry, I have a little Ruby experience. I cannot really understand how to split the arguments for the script and for the Cocoapods command without specifying all Cocoapods options in `OptionParser`. Would you give me a hint please?

Reply:
I'm not sure how to do this with the `OptionParser` but another approach would be to pass the `--ignore-local-specs` as an additional ignore option to `find_local_deps` and skip the `results.push` if the `dep_podspec` is in the ignore list.

Reply:
@paulb777 it is basically what the script does right now. The arg `--ignore-local-specs` has to be removed from the `pod` args, otherwise it fails with `unknown option` error.

Reply:
Oh I see. Apologies, I didn't read this closely enough to realize you were only parsing a subset of parameters. This turns out not to be easy to do with Ruby's `OptionParser` (it is with Python's). Here's some feedback instead:

This code can be considerably simpler:
  * just use a regular if/else instead of case
  * use groups in your regular expression and use that to extract the part of the flag you want to keep

Also:
  * `Array.push` is the canonical way to add to the end of an array. I highlight this only because this is inconsistent with usage of that below
  * Prefer the parenthesis form of function invocation for consistency with other usages in this file.
  * You might consider rewording `local_specs_to_ignore` to just mirror the argument name (i.e. to `ignore_local_pecs`)
  * You can split that value directly to avoid needing both `local_specs_to_ignore_arg` and `local_specs_to_ignore`.

```ruby
ignore_local_specs = []

args.each do |arg|
  if arg =~ /--ignore-local-specs=(.*)/
    ignore_local_specs = $1.split(',')
  else
    pod_args.push(arg)
  end
end
```

Reply:
Great, thank you for the detailed explanation! I updated the code accordingly.